src/main.js

```diff
@@ -42,23 +51,47 @@ import { TrajectoryCalculator } from "./core/trajectoryCalculator.js";
 
     // Checking ground collision
     await activeGround.isThereCollision(playerOne);
-    let isFalling = true;
+    await activeGround.isThereCollision(playerTwo);
+    let [isPlayerOneFalling, isPlayerTwoFalling] = [true, true];
+    let playerTurn = true;
+    let [playerOneMoveDist, playerTwoMoveDist] = [20, 20];
 
     // Create ticker in order to update sprite positioning
     app.ticker.add(() => {
-        playerOne.updatePlayerPosition();
         playerOne.updateBullets();
-        if (playerOne.checkSpaceBarInput()) {
+        playerTwo.updateBullets();
+        if (playerOne.checkSpaceBarInput() && playerTurn) {
             playerOne.createBullet();
+        } else if (playerTwo.checkSpaceBarInput() && !playerTurn) {
+            playerTwo.createBullet();
         }
 
-        let isColliding = activeGround.isThereCollision(playerOne);
-        if (isColliding){
-            isFalling = false;
-        };
-
-        if (isFalling){
+        // Ground collision and movement detection
+        playerOne.updatePlayerPosition();
+        playerTwo.updatePlayerPosition();
+        activeGround.isThereCollision(playerOne);
+        if (isPlayerOneFalling){
             playerOne.applyGravity();
-        };
+        }
+        activeGround.isThereCollision(playerTwo);
+        if (isPlayerTwoFalling){
+            playerTwo.applyGravity();
+        }
+
```

Comment:
maybe add comment here to say what the piece of code does below

Reply:
Respectfully no